### PR TITLE
Pedigree c infrastructure + use parents column

### PR DESCRIPTION
**Pull request by jeromekelleher**
Some basic C infrastructure here @nikbaya, and also the minimal changes needed to use the ``parent`` column in the Individual table.

I've left the Python side broken for now as there's a fair bit to be done infrastructure wise and we might as well clean things up there first.

### CODE REVIEW

File: lib/tests/testlib.c [hunk 1]
```diff
@@ -78,9 +78,10 @@ build_pedigree_sim(msp_t *msp, tsk_table_collection_t *tables, gsl_rng *rng,
 
     /* Insert the pedigree individuals */
     for (j = 0; j < num_individuals; j++) {
-        ret = tsk_individual_table_add_row(&tables->individuals, 0, NULL, 0, NULL, 0,
-            /* encode the parents in the metadata */
-            (void *) (parents + j * ploidy), ploidy * sizeof(tsk_id_t));
+        ret = tsk_individual_table_add_row(
+            &tables->individuals, 0, NULL, 0, parents + j * ploidy, ploidy, NULL, 0);
+        /* NULL, 0, */
```
Comment (benjeffery) [inline lib/tests/testlib.c:83]: I guess you're keeping this here for future ref? Maybe a note to that effect if that is the intent.

**Review by benjeffery — APPROVED** [review]
LGTM, one question about a comment.
So great to see the clean up from getting the parents in the data model properly!

File: lib/tests/testlib.c (hunk 1 above)
Comment (jeromekelleher) [reply lib/tests/testlib.c:83]: Hah, no, I just missed it. Good spot.

**Comment by jeromekelleher** [comment]
Not for merging yet, needs some cleanup.

**Pull request merged**